### PR TITLE
Add snapshot tests for iOS 17

### LINKED ISSUE #155: Add snapshot tests for iOS 17 and audit results
* Add an iOS 17 device to the list in `SnapshotTestCase` and record new reference images. Make sure to choose a simulator that's included in the [virtual image](https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md#installed-simulators) for our CI jobs.
* Update the build scripts and CI config to include an iOS 17 build
* Audit the reference images against VoiceOver on a device running iOS 17 and file issues for any changes/regressions where the behavior doesn't match

### PR DESCRIPTION
**Pull request by NickEntin**
Resolves #155

### CODE REVIEW

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilityContainersTests/testDataTableWithHeaders_17_2_393x852@3x.png
Comment (NickEntin) [inline Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilityContainersTests/testDataTableWithHeaders_17_2_393x852@3x.png:1]: Data table descriptions are still incorrect on iOS 17, see #7

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testButtonTraits_17_2_393x852@3x.png
Comment (NickEntin) [inline Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testButtonTraits_17_2_393x852@3x.png:1]: 16 now has a hint "Use the rotor to enable direct touch for this app"

Comment (NickEntin) [inline Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testButtonTraits_17_2_393x852@3x.png:1]: 6 and 22 now read the description as listed here, then immediately repeat the description without "image"

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testSwitchControls_17_2_393x852@3x.png
Comment (NickEntin) [inline Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testSwitchControls_17_2_393x852@3x.png:1]: The last three now read:

"Fake label, Switch button, on" / "Double tap to toggle setting."
"Fake label, Switch button 3" / "Double tap to toggle setting."
"Fake label, Switch button value" / "Double tap to toggle setting."

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testTabBars_17_2_393x852@3x.png
Comment (NickEntin) [inline Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testTabBars_17_2_393x852@3x.png:1]: There's a few differences in this one. Interestingly in the second row Item B now includes "3 items" in the description. Updated #26 with these findings.

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.ActivationPointTests/testActivationPointEnabledWhenOverridden_17_2_393x852@3x.png
Comment (NickEntin) [inline Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.ActivationPointTests/testActivationPointEnabledWhenOverridden_17_2_393x852@3x.png:1]: Looks like activation points are broken again 😞

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.DefaultControlsTests/testStepperAtMin_17_2_393x852@3x.png
Comment (NickEntin) [inline Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.DefaultControlsTests/testStepperAtMin_17_2_393x852@3x.png:1]: These should still say "decrement" and "increment"

File: Example/SnapshotTests/__Snapshots__/SnapshotTestingTests/testShowingActivationPoint.always-393x852-17-2-3x.png
Comment (NickEntin) [inline Example/SnapshotTests/__Snapshots__/SnapshotTestingTests/testShowingActivationPoint.always-393x852-17-2-3x.png:1]: Broken activation points for SnapshotTesting too

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.ActivationPointTests/testActivationPointEnabledWhenOverridden_17_2_393x852@3x.png
Comment (NickEntin) [reply Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.ActivationPointTests/testActivationPointEnabledWhenOverridden_17_2_393x852@3x.png:1]: Nevermind, looks like this is just an issue with my machine

File: Example/SnapshotTests/__Snapshots__/SnapshotTestingTests/testShowingActivationPoint.always-393x852-17-2-3x.png
Comment (NickEntin) [reply Example/SnapshotTests/__Snapshots__/SnapshotTestingTests/testShowingActivationPoint.always-393x852-17-2-3x.png:1]: Likewise issue with my machine

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.DefaultControlsTests/testStepperAtMin_17_2_393x852@3x.png
Comment (NickEntin) [reply Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.DefaultControlsTests/testStepperAtMin_17_2_393x852@3x.png:1]: From what I can tell this is somewhat flaky. Not sure what's going on here. Same root problem as #29, but maybe less common. I'll keep an eye on it.

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testButtonTraits_17_2_393x852@3x.png
Comment (NickEntin) [reply Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testButtonTraits_17_2_393x852@3x.png:1]: I don't think this needs to be reflected in the snapshot. It's odd behavior, but what's here is still the data we want to represent.

Comment (NickEntin) [reply Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testButtonTraits_17_2_393x852@3x.png:1]: I think this is related to how you have direct touch configured, so I'm gonna ignore this for now

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testTabBars_17_2_393x852@3x.png
Comment (NickEntin) [reply Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testTabBars_17_2_393x852@3x.png:1]: Interestingly this one got fixed in the CI snapshot updates. Maybe it's flaky?

File: Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testSwitchControls_17_2_393x852@3x.png
Comment (NickEntin) [reply Example/SnapshotTests/ReferenceImages/_64/SnapshotTests.AccessibilitySnapshotTests/testSwitchControls_17_2_393x852@3x.png:1]: Filed as #199

**Review by kruegermj — APPROVED** [review]

**Pull request merged**